sql/analyzer/validation_rules.go

```diff
@@ -211,6 +211,26 @@ func validateGroupBy(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, se
 	span, ctx := ctx.Span("validate_group_by")
 	defer span.End()
 
+	// only enforce strict group by when this variable is set
+	var strictGroupBy bool
+	if sysVar, sysVal, ok := sql.SystemVariables.GetGlobal("sql_mode"); ok {
```

Comment:
`sql_mode` can also be set for a session, so probably makes sense to have this pull from the session, instead of just the global setting.